src/acl_auto_configure.cpp

```diff
@@ -493,6 +493,61 @@ bool acl_load_device_def_from_str(const std::string &config_str,
                                 counters);
   }
 
+  // Read device global information.
+  unsigned int num_device_global = 0;
+  if (result && counters.back() > 0) {
+    result =
+        read_uint_counters(config_str, curr_pos, num_device_global, counters);
+    devdef.num_device_global = num_device_global;
+
+    // read total number of fields in device global
+    int total_fields_device_global = 0;
+    if (result) {
+      result = read_int_counters(config_str, curr_pos,
+                                 total_fields_device_global, counters);
+    }
+
+    for (auto i = 0U; result && (i < num_device_global);
+         i++) { // device_global_memories
+      counters.emplace_back(total_fields_device_global);
+
+      // read device global name
+      std::string device_global_name;
+      if (result && counters.back() > 0) {
+        result = read_string_counters(config_str, curr_pos, device_global_name,
+                                      counters);
+      }
+
+      // read device global address
+      unsigned int dev_global_addr = 0; // Default
+      if (result && counters.back() > 0) {
+        result =
+            read_uint_counters(config_str, curr_pos, dev_global_addr, counters);
+      }
+      // read device global address size
+      unsigned int dev_global_size = 0; // Default
+      if (result && counters.back() > 0) {
+        result =
+            read_uint_counters(config_str, curr_pos, dev_global_size, counters);
+      }
+
+      acl_device_global_mem_def_t dev_global_def = {
+          device_global_name, dev_global_addr, dev_global_size};
+      devdef.device_global_mem_defs[device_global_name] = dev_global_def;
```

Comment:
If the key does not exist yet, this will insert a default-constructed value and then overwrite with the given value. If the key does exist, this will overwrite the existing value. You can use [`std::unordered_map::insert`](https://en.cppreference.com/w/cpp/container/unordered_map/insert) to avoid the default construction when the key does not exist, and to signal an error when the key does exist.